tests/OmniSharp.Tests/NavigateUpDownFacts.cs

```diff
@@ -0,0 +1,751 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using OmniSharp.Filters;
+using OmniSharp.Models;
+using Xunit;
+
+namespace OmniSharp.Tests
+{
+    enum NavigateDirection 
+    {
+      UP,
+      DOWN
+    }
+
+    public class NavigateUpDownFacts
+    {
+
+        private string foo = "test";
+
+        private async Task<NavigateResponse> SendRequest(OmnisharpWorkspace workspace, string fileName, string fileContent, NavigateDirection upOrDown)
+        {
+            var initialCursorLineColumn = TestHelpers.GetLineAndColumnFromDollar(TestHelpers.RemovePercentMarker(fileContent));
+            var fileContentNoMarker = TestHelpers.RemovePercentMarker(TestHelpers.RemoveDollarMarker(fileContent));
+            var controller = new OmnisharpController(workspace, null);
+            var request = new Request
+            {
+                Line = initialCursorLineColumn.Line,
+                Column = initialCursorLineColumn.Column,
+                FileName = fileName,
+                Buffer = fileContentNoMarker
+            };
+
+            if (upOrDown == NavigateDirection.UP) 
+            {
+                return await controller.NavigateUp(request);
+            }
+            else 
+            {
+                return await controller.NavigateDown(request);
+            }
+        }
+
+        private async Task AssertPosition(string fileContent, NavigateDirection navigateDirection, int finalCursorColumnOffset = 1) 
```

Comment:
I don't see why this is needed. 

Should always be 0 or some constant. It shouldn't ever change. Looks as if it's masking some bug in the test code, because I can see that this code works.